src/abstractinterpret/typeinfer.jl

```diff
@@ -150,6 +150,10 @@ end
 # global
 # ------
 
+@static if VERSION ≥ v"1.11.0-DEV.1552"
+CC.cache_owner(analyzer::AbstractAnalyzer) = AnalysisCache(analyzer)
```

Comment:
I am still not sure if `AnalysisCache` is the right object here. What needs to be captured is information that changes inference behaviour. See https://github.com/JuliaGPU/GPUCompiler.jl/blob/8d7d9836447c067973a2c90aa4b274a09e7b2b20/src/interface.jl#L223 as an alternative idea.

Reply:
Actually, `AnalysisCache(analyzer)` is unique, kind of like how `GPUCompilerCacheToken` is. The reason is each analyzer computes unique hash based on analysis settings, creating different `AnalysisCache` objects for different settings: https://github.com/aviatesk/JET.jl/blob/56520e0615079f1d65fb4094eec11e740773afdd/src/analyzers/jetanalyzer.jl#L77